File: ksa_vat/doc_events/item.py
import frappe
from frappe.model.document import Document
from frappe import _


def on_update(doc, method):
    if not doc.custom_item_tax_template:
        return

    tax_template = frappe.get_doc('Tax Template', doc.custom_item_tax_template)

    existing_templates = { row.item_tax_template for row in doc.taxes }

    for tax_row in tax_template.taxes:
        if tax_row.item_tax_template not in existing_templates:
            doc.append('taxes', {
                'item_tax_template': tax_row.item_tax_template,
                'tax_category': tax_row.tax_category,
                'valid_from': tax_row.valid_from,
                'minimum_net_rate': tax_row.minimum_net_rate,
                'maximum_net_rate': tax_row.maximum_net_rate
            })

File: ksa_vat/public/js/item.js
frappe.ui.form.on("Item", {
    custom_item_tax_template: function(frm) {
        if (frm.doc.custom_item_tax_template) {
            frappe.call({
                method: "frappe.client.get",
                args: {
                    doctype: "Tax Template",
                    name: frm.doc.custom_item_tax_template
                },
                callback: function(r) {
                    if (r.message) {
                        frm.clear_table("taxes");

                        $.each(r.message.taxes || [], function(i, tax) {
                            let row = frm.add_child("taxes");
                            row.item_tax_template = tax.item_tax_template;
                            row.tax_category = tax.tax_category;
                            row.valid_from = tax.valid_from;
                            row.minimum_net_rate = tax.minimum_net_rate;
                            row.maximum_net_rate = tax.maximum_net_rate;
                        });

                        frm.refresh_field("taxes");
                    }
                }
            });
        }
        else {
            frm.clear_table('taxes');
            frm.refresh_field('taxes');
        }
    }
    
});
